internal/controller/controllers/controller_event_wrapper.go

```diff
@@ -0,0 +1,47 @@
+package controllers
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/jinzhu/gorm"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/events"
+	"github.com/sirupsen/logrus"
+)
+
+type controllerEventsWrapper struct {
+	events           *events.Events
+	crdEventsHandler CRDEventsHandler
+	db               *gorm.DB
+	log              logrus.FieldLogger
+}
+
+func NewControllerEventsWrapper(crdEventsHandler CRDEventsHandler, events *events.Events, db *gorm.DB, log logrus.FieldLogger) *controllerEventsWrapper {
+	return &controllerEventsWrapper{crdEventsHandler: crdEventsHandler,
+		events: events, db: db, log: log}
+}
+
+func (c *controllerEventsWrapper) AddEvent(ctx context.Context, clusterID strfmt.UUID, hostID *strfmt.UUID, severity string, msg string, eventTime time.Time) {
+	c.events.AddEvent(ctx, clusterID, hostID, severity, msg, eventTime)
+	cluster, err := common.GetClusterFromDB(c.db, clusterID, common.SkipEagerLoading)
+	if err != nil {
+		return
+	}
+
+	// Notifying cluster deployment in case host id is nil -> cluster event
+	if hostID == nil {
+		c.log.Debugf("Pushing cluster event %s %s", cluster.KubeKeyName, cluster.KubeKeyNamespace)
```

Comment:
it is not periodic so maybe it can be an info

Reply:
it will spawn too much logs

Reply:
and will be rewritten in couple of days so no point